packages/liveblocks-react/src/__tests__/index.test.tsx

```diff
@@ -414,3 +437,56 @@ describe("useStorage", () => {
     expect(render1).toBe(render2); // Referentially equal!
   });
 });
+
+describe("useCanUndo / useCanRedo", () => {
+  test("can undo and redo", async () => {
+    const canUndo = renderHook(() => useCanUndo());
+    const canRedo = renderHook(() => useCanRedo());
+    const undo = renderHook(() => useUndo());
+    const mutation = renderHook(() =>
+      useMutation(({ root }) => root.get("obj").set("a", Math.random()))
+    );
+
+    expect(canUndo.result.current).toEqual(false);
+    expect(canRedo.result.current).toEqual(false);
+
+    const sim = await websocketSimulator();
+    act(() => sim.simulateExistingStorageLoaded());
+
+    expect(canUndo.result.current).toEqual(false);
+    expect(canRedo.result.current).toEqual(false);
+
+    // Run a mutation
+    act(() => mutation.result.current());
+
+    expect(canUndo.result.current).toEqual(true);
+    expect(canRedo.result.current).toEqual(false);
+
+    // Undo that!
+    act(() => undo.result.current());
+
+    expect(canUndo.result.current).toEqual(false);
+    expect(canRedo.result.current).toEqual(true);
+
+    // Run 3 mutations
+    act(() => mutation.result.current());
+    act(() => mutation.result.current());
+    act(() => mutation.result.current());
+
+    expect(canUndo.result.current).toEqual(true);
+    expect(canRedo.result.current).toEqual(false);
+
+    // Undo 2 of them
+    act(() => undo.result.current());
+    act(() => undo.result.current());
+
+    expect(canUndo.result.current).toEqual(true);
+    expect(canRedo.result.current).toEqual(true);
+
+    // Undo the last one
+    act(() => undo.result.current());
+
+    expect(canUndo.result.current).toEqual(false);
+    expect(canRedo.result.current).toEqual(true);
+  });
+});
```

Comment:
Also added some unit tests for these (which is how I discovered the bug from #474).